data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -22,17 +47,155 @@
+            properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+                    StringDeserializer.class);
+            properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, JsonDeserializer.class);
+        } else if (schemaType.equalsIgnoreCase(MessageFormat.PLAINTEXT.toString())) {
+            properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+                    StringDeserializer.class);
+            properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+                    StringDeserializer.class);
+        } else if (schemaType.equalsIgnoreCase(MessageFormat.AVRO.toString())) {
+            properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+                    StringDeserializer.class);
+            properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+                    KafkaAvroDeserializer.class);
+            properties.put(AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, sourceConfig.getSchemaConfig().getRegistryURL());
+        }
+    }
 
-  @Override
-  public void stop() {
-   LOG.info("Initiated the stop method...");
-  }
+    private void setPropertiesForAuth(Properties properties) {
+        String username = sourceConfig.getAuthConfig().getPlainTextAuthConfig().getUsername();
+        String password = sourceConfig.getAuthConfig().getPlainTextAuthConfig().getPassword();
+        properties.put("sasl.mechanism", "PLAIN");
+        properties.put("sasl.jaas.config", "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"" + username + "\" password=\"" + password + "\";");
+        properties.put("security.protocol", "SASL_PLAINTEXT");
+    }
 
+    private static String getSchemaType(final String registryUrl, final String topicName, final int schemaVersion) {
+        StringBuilder response = new StringBuilder();
+        String schemaType = "";
+        try {
+            String urlPath = registryUrl + "subjects" + "/" + topicName + "-value/versions/" + schemaVersion;
+            URL url = new URL(urlPath);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
+            int responseCode = connection.getResponseCode();
+            if (responseCode == HttpURLConnection.HTTP_OK) {
+                BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+                String inputLine;
+                while ((inputLine = reader.readLine()) != null) {
+                    response.append(inputLine);
+                }
+                reader.close();
+                ObjectMapper mapper = new ObjectMapper();
+                Object json = mapper.readValue(response.toString(), Object.class);
+                String indented = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(json);
+                JsonNode rootNode = mapper.readTree(indented);
+                if (rootNode.has("schemaType")) {
```

Comment:
this doesn't look very reliable way of determining schemaType. is there not other way which is more deterministic? also protobuf is also supported schema type by registries.

Reply:
This is the only way we get response back from the schema registry and based on that, we try to find out the schema type. Protobuf is not in scope now.

Reply:
I think you should check for presence of `schemaType` and if it is not present, make it AVRO, if it is present, it should be taken from the value. If the contents has `schemaType: AVRO`, the current code would make it JSON. This will handle JSON and PROTOBUF(when it is eventually supported)